client/src/components/RepoSettings.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { RepoExtDTO } from '../api/repo.api';
+import { Card, Button, Alert } from 'react-bootstrap';
+import { RepositoryService } from '../api/repo.api';
+import { ToastType, useToastStore } from '../util/toastStore';
+import { AxiosError } from 'axios';
+import { RepositoryVisibilityUpdateDTO } from '../api/repo.api' 
+
+export const RepoSettings: React.FC<{ isActive: boolean, repo: RepoExtDTO, repoStateChanger: any }> = (props) => {
+    const addToast = useToastStore((state) => state.addToast);
+    const [error, setError] = useState('');
+
+    useEffect(() => {
+        if (props.isActive) {
+            // Tab selected, do stuff here...
+        }
+    }, [props.isActive]);
+
+    const changeVisibility = () => {
+        let dto: RepositoryVisibilityUpdateDTO = {
+            public: !props.repo.public,
+        };
+
+        RepositoryService.UpdateRepoVisibilityById(props.repo.id, dto).then((res) => {
+            props.repoStateChanger({
+                ...props.repo,
+                ...res.data,
+            });
+            addToast(`Updated repository ${props.repo.name}`, ToastType.success);
+            setError('');
```

Comment:
Since this function _exclusively_ updates the visibility of the repository, make the toast message more descriptive as well ("{reponame} is now a private repository" or something like that).